File: Y2020/01/part2.c
#include <stdio.h>

int main() {
    FILE *fp = fopen("data.txt", "r");
    int nums[1024];
    int count = 0;
    int num;
    while(fscanf(fp, "%d\n", &num) == 1) {
        nums[count++] = num;
    }
    printf("%d\n", count);
    for(int i = 0; i < count - 1; i++) {
        for(int j = i + 1; j < count; j++) {
            for(int k = j + 1; k < count; k++) {
		if(nums[i] + nums[j] + nums[k] == 2020) {
		        printf("%d\n", nums[i] * nums[j] * nums[k]);
		        return 0;
		}
	     }
	}
    }
    return 0;
}

File: Y2023/D04/part2.js
const fs = require('fs')

const input = fs.readFileSync(`${__dirname}/data.txt`, 'utf8').split('\n')

const cardCounts = {};
for(const card of input) {
  const [_, cardNum, winNumStr, numStr] = card.match(/Card\W+(\d+): ([\d ]+) \| ([\d ]+)/);
  const winNums = winNumStr.replaceAll('  ', ' ').trim().split(' ').map(Number);
  const nums = numStr.replaceAll('  ', ' ').trim().split(' ').map(Number);
  cardCounts[cardNum] ??= 0;
  cardCounts[cardNum]++;
  let t = 0;
  for(const num of nums) {
    if(winNums.includes(num)) {
      t++;
    }
  }
  for(let i = +cardNum + 1; i < +cardNum + 1 + t; i++) {
    cardCounts[i] ??= 0;
    cardCounts[i] += cardCounts[cardNum];
  }
}
console.log(Object.values(cardCounts).reduce((a, b) => a + b, 0));